### DISCUSSION TITLE
Firebase Hostingで動的ページを生成するNextアプリがデプロイできない

### DISCUSSION THREAD

**Discussion by Yukitkym** [if-mentor/next_fire_todo_5#96] [Q&A]
【起こっている問題】
デプロイをした後デプロイ先のURLにアクセスすると、firebase hosting setup completeと出てくるもののTodoアプリの内容が表示されない

【原因】
buildを行った際のページ構成がFirebase Hostingが読み込みたいページ構成とマッチしていない
ビルド先のファイルの直下にindex.htmlがなく、firebase deployする際に自動生成されたindex.htmlが表示されてしまっている
https://yoheiko.com/blog/2021/01/23/post-769/#outline_1__4

【やったこと】
・next.config.jsにdistDir: 'build'を追記し、.nextではなくbuildにビルド結果が追加されるようにした
https://nextjs-ja-translation-docs.vercel.app/docs/api-reference/next.config.js/setting-a-custom-build-directory
→ビルド先は変わったが、状況は変わらず

・buildしてexportを行い、outに静的ページを作成
https://stackoverflow.com/a/58162910
→静的なページは表示されるようになったが、動的なページ(/todo/[id]など)は表示されない

【現在の状況】
静的なページは表示されるようになったが、動的なページ(/todo/[id]など)は表示されない
https://todo-team5-next.web.app/

【解決策】
①Vercelにデプロイをする
②なんとか動的なページをFirebase Hostingで実装する方法を探す

＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿
Firebase Hostingでデプロイするのは諦め、Vercelにデプロイするのが良いと思うのですが、皆さんいかがでしょうか？

Vercelにデプロイする場合は、Git Hubアカウントとの連携が必要になるので、ジュンペイさんに今後デプロイをお願いすることになりそうです。(buildまでは自分達でやる)
もしくは、チーム開発のGit HubのRepositoryをprivateからpublicに変更すれば、誰でもデプロイできるようになるかもしれません。

**Comment by Junkichi89** [comment]
@Yukitkym 

この辺参考にしてみてもうまくいかないでしょうか？
https://code-log.hatenablog.com/entry/2020/02/23/200541

**Reply by Yukitkym** [reply]
@Junkichi89 
静的サイトとしてのデプロイはできたのですが、Todo詳細に飛ぼうとすると/todo/[id]のような動的なページ生成ができないので、404ページに飛んでしまうんですよね。

**Reply by Junkichi89** [reply]
結論から言うと、todo/[id].tsxの修正が必要かと思います。

おそらく原因は、こちらの記述ですね！
![image](https://user-images.githubusercontent.com/65665381/172391022-483267aa-c55f-49e4-bdff-99107ca03203.png)

今回の開発では、CSR(クライアントサイドレンダリング)で実装を行う必要があるため、SSGの機能は使用しない形で実装を進めていく必要がありますね。
クライアントでデータを取得するような形の実装に修正しましょう！

**Reply by Yukitkym** [reply]
@Junkichi89 
ご指摘ありがとうございます。
ご指摘の通り、そちらが原因でした。

『Todo詳細に飛ぼうとすると/todo/[id]のような動的なページ生成ができない』、こちらのエラーはTodo詳細ページがFirebase Hosting関係なしにローカル環境でも発生していた問題でした。
確認不足でディスカッションに送ってしまい、申し訳ございません。

おかげさまで問題なくFirebase Hostingでもデプロイできました。
ありがとうございます。🙇
